main.cpp

```diff
@@ -1 +1,215 @@
+                {0,0,-1},
+                {0,-1,0},
+                {-1,0,0},
+        };
+/*
+ * Affiche le tableau de jeu dans la console
+ */
+void PrintArray()
+{
+    std::cout << "---------------" << std::endl;
+    for(auto & i : board)
+    {
+        std::cout << "|";
+        for(signed char j : i)
+        {
+            if(j == -1) std::cout << " X ";
+            else if(j == 1) std::cout << " O ";
+            else std::cout << "   ";
+            std::cout << "|";
+        }
+        std::cout << std::endl;
+    }
+    std::cout << "---------------" << std::endl;
+}
+
+/*
+ * Remplis le tabelau de jeu avec des valeurs aléatoires de -1, 0 ou 1
+ */
+void GenerateRandomArray()
+{
+    srand(time(0));
+    for(auto & i : board)
+    {
+        for(signed char & j : i)
+        {
+            j = (rand()%3) - 1;
+        }
+    }
+}
+
+/*
+ * Initialise le tableau en remplissant toutes ses cases avec un 0
+ */
+void InitializeArray()
+{
+    for(auto & i : board)
+    {
+        for(signed char & j : i)
+        {
+            j = 0;
+        }
+    }
+}
+
+/*
+ * Gère le tour de jeu pour un joueur humain
+ */
+void HumanPlay(bool isCrossPlaying)
+{
+    std::string cell;
+    uint8_t xCoord, yCoord;
+    bool stayInLoop = true;
+
+    isCrossPlaying ? std::cout << "Cross is playing" << std::endl : std::cout << "Circle is playing" << std::endl;
+
+    do {
+        std::cin >> cell;
+        xCoord = cell[0] - '0';
+        yCoord = cell[1] - '0';
+        if(cell.length() == 2)
+        {
+            if(xCoord <= SIZE && xCoord > 0 && yCoord <= SIZE && yCoord > 0)
+            {
+                //std::cout << "Played on cell " + std::to_string((xCoord*10) + yCoord) << std::endl;
+                if(board[xCoord][yCoord] == 0)
+                {
+                    board[xCoord][yCoord] = isCrossPlaying ? -1 : 1;
+                    stayInLoop = false;
+                }
+                else
+                {
+                    std::cout << "You can't play on a non-empty cell. Please play again." << std::endl;
+                }
+            }
+            else
+            {
+                std::cout << "cell[0] = " + std::to_string(cell[0]) + " and cell[1] = " + std::to_string(cell[1]) << std::endl;
+                std::cout << "You can't enter coordinates below 0 or above the size of the playing board (" + std::to_string(SIZE) + ")." << std::endl;
+            }
+        }
+        else
+        {
+            std::cout << "Please enter only to numbers (x and y coordinate) ranging from 1 to " + std::to_string(SIZE) + "." << std::endl;
+        }
+    } while (stayInLoop);
+}
+
+/*
+ * Gère un tour de jeu pour une IA
+ */
+void AIPlay(bool isCrossPlaying)
+{
+
+}
+
+/*
+ * Parcours le tableau et détermine s'il y a un gagnant.
+ */
+uint8_t CheckForWinner()
+{
+    bool equality = true;
+    int8_t row = 0,column = 0,diagonal1 = 0, diagonal2 = 0;
+
+    for(auto & i : board)
+    {
+        for(uint8_t j=0;j<SIZE;j++)
```

Comment:
I prefer `auto`

Reply:
```suggestion
        for(auto j: i)
```

That should work, no? I didn't test it.